cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -93,38 +94,74 @@ private HttpClient createHttpClient(String url) {
 
   @Override
   public void download(String url, Path target) {
+    List<String> httpProtocols = IdeVariables.HTTP_PROTOCOLS.get(context);
+    List<HttpClient.Version> protocolVersions = new ArrayList<>();
+
+    for (String httpProtocol : httpProtocols) {
+      switch (httpProtocol) {
+        case "HTTP_1_1":
+          protocolVersions.add(HttpClient.Version.HTTP_1_1);
+          this.context.debug("Added HTTP_1_1 protocol");
+          break;
+        case "HTTP_2":
+          protocolVersions.add(HttpClient.Version.HTTP_2);
+          this.context.debug("Added HTTP_2 protocol");
+          break;
+        default:
+          this.context.error("Unsupported protocol was set in HTTP_PROTOCOLS: " + httpProtocol);
```

Comment:
looks a little redundant/boilder-plate...
Nothing wrong, but we already have `VariableDefinitionEnum` (not yet supporting lists) or you could handle this generic via the Enum here...